src/resnext.jl

```diff
@@ -0,0 +1,137 @@
+        conv_bn((3, 3), outplanes[1], outplanes[2]; stride = 1, pad = 1,
+                    groups = cardinality, bias = false)...,
+        conv_bn((1, 1), outplanes[2], outplanes[3], identity; stride = 1, bias = false)...)
+
+"""
+    resnext(connection = (x, y) -> @. relu(x) + relu(y); channel_config, block_config, 
+            cardinality, nclasses = 1000)
+    
+Create a ResNeXt model
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments
+- `connection`: the binary function applied to the output of residual and skip paths in a block
+- `channel_config`: the growth rate of the output feature maps within a residual block
+- `block_config`: a list of the number of residual blocks at each stage
+- `cardinality`: the number of groups to use for the convolution
+- `nclasses`: the number of output classes
+"""
+function resnext(connection = (x, y) -> @. relu(x) + relu(y); channel_config, block_config, cardinality, nclasses = 1000)
+    inplanes = 128
+    baseplanes = 128
+    layers = []
+    append!(layers, conv_bn((7, 7), 3, inplanes; stride = 2, pad = (3, 3)))
+    push!(layers, MaxPool((3, 3), stride = (2, 2), pad = (1, 1)))
+    for (i, nrepeats) in enumerate(block_config)
+        # output planes within a block
+        outplanes = baseplanes .* channel_config
+        # push first skip connection on using first residual
+        # downsample the residual path if this is the first repetition of a block
+        push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, i != 1),
+                                            skip_projection(inplanes, outplanes[end], i != 1)))
+        # push remaining skip connections on using second residual
+        inplanes = outplanes[end]
+        for _ in 2:nrepeats
+            push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, false),
+                                                skip_identity(inplanes, outplanes[end], false)))
+            inplanes = outplanes[end]
+        end
+        # next set of output plane base is doubled
+        baseplanes *= 2
+    end
+
+    return Chain(Chain(layers...),
+                Chain(AdaptiveMeanPool((1, 1)), flatten, Dense(inplanes, nclasses)))
+end
+
+"""
+    ResNeXt(channel_config, block_config, cardinality; nclasses = 1000)
+    
+Create a ResNeXt model
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments
+- `channel_config`: the growth rate of the output feature maps within a residual block
+- `block_config`: a list of the number of residual blocks at each stage
+- `cardinality`: the number of groups to use for the convolution
+- `nclasses`: the number of output classes
+"""
+struct ResNeXt
+    layers
+end
+
+function ResNeXt(channel_config, block_config, cardinality; nclasses = 1000)
+    layers = resnext(;channel_config = channel_config,
+                  block_config = block_config,
+                  cardinality = cardinality,
+                  nclasses = nclasses)
```

Comment:
Seems to me that the API is backwards here. `channel_config`, `block_config` and `cardinality` need to be positional, and have convenience constructors for skipping the `connection`.

Reply:
That's because this mirrors the `resnet` API (as it should for consistency) and the `resnet` API contains additional positional arguments that are more important, but those arguments are dropped here. In any case, I think the API needs some more significant changes since ResNeXt is specified in terms of cardinality and width, not a channel configuration. I've suggested an alternate API where the relevant positional args are given first.